Tests/ClientTests/Wallpaper/WallpaperMetadataTrackerTests.swift

```diff
@@ -0,0 +1,76 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Shared
+
+import XCTest
+
+@testable import Client
+
+class WallpaperMetadataTrackerTests: XCTestCase, WallpaperJSONTestProvider {
+
+    // MARK: - Properties
+    var sut: WallpaperMetadataUtility!
+    var mockUserDefaults: MockUserDefaults!
+    var mockNetwork: NetworkingMock!
+
+    private let prefsKey = PrefsKeys.Wallpapers.MetadataLastCheckedDate
+
+    // MARK: - Setup & Teardown
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+        mockUserDefaults = MockUserDefaults()
+        mockNetwork = NetworkingMock()
+        sut = WallpaperMetadataUtility(with: mockNetwork,
+                                       and: mockUserDefaults)
+    }
+
+    override func tearDownWithError() throws {
+        sut = nil
+        mockUserDefaults = nil
+        mockNetwork = nil
+        try super.tearDownWithError()
+    }
+
+    // MARK: Tests
+    func testMetadataTracker_whenInitializedFirstTime_fetchesFreshData() async {
+//        setupNetwork(for: .goodData)
```

Comment:
nit: Can be removed? or it will be fix in the future

Reply:
Will be fixed in the future for this one.